File: 2.Valid Anagram/solution.cpp
#include <iostream>
#include <string>
#include <unordered_map>

class Solution {
public:
    bool isAnagram(std::string s, std::string t) {
        if (s.length() != t.length()) return false;
        std::unordered_map<char, int> countS, countT;
        for (char c : s) countS[c]++;
        for (char c : t) countT[c]++;
        return countS == countT;
    }
};

// Example usage
int main() {
    Solution solution;
    std::cout << solution.isAnagram("anagram", "nagaram") << std::endl;  // Output: 1 (true)
    std::cout << solution.isAnagram("rat", "car") << std::endl;  // Output: 0 (false)
    return 0;
}

File: 2.Valid Anagram/solution.java
import java.util.HashMap;

public class solution {
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length()) return false;
        HashMap<Character, Integer> countS = new HashMap<>();
        HashMap<Character, Integer> countT = new HashMap<>();
        for (char c : s.toCharArray()) countS.put(c, countS.getOrDefault(c, 0) + 1);
        for (char c : t.toCharArray()) countT.put(c, countT.getOrDefault(c, 0) + 1);
        return countS.equals(countT);
    }

    // Example usage
    public static void main(String[] args) {
        solution solution = new solution();
        System.out.println(solution.isAnagram("anagram", "nagaram"));  // Output: true
        System.out.println(solution.isAnagram("rat", "car"));  // Output: false
    }
}
